EXTENDING_REMIXER.md

```diff
@@ -1 +1,76 @@
-TODO: This page is a placeholder. Details on how to extend Remixer through adding data types need to be added here.
\ No newline at end of file
+# Extending Remixer
+
+There are several ways in which you can extend remixer:
+
+- Add a new `DataType`:
+  - Create a `ValueConverter` for the `DataType`
+  - Set as default layoutIDs by calling `DataType.setLayoutIdForVariableType`
+  - Register it at `Application.onCreate` time
+    - can be done in `RemixerInitialization.initRemixer` if you're adding it at the Remixer level (forking/contributing)
+  - **Note**: At the time only forks/contributions can add annotation processing support for new DataTypes. If it proves necessary we'll write extension points for the annotation processor. 
+- Implement a new `RemixerWidget`
+  - Optionally, set it as a default Layout Id for a `DataType`/`Variable class` combination by calling `DataType.setLayoutIdForVariableType`. You can do it in `RemixerInitialization.initRemixer` if forking or contributing to Remixer.
+
+**Notice that if you plan to use the Firebase Remote Controller functionality, you need to replicate this work in the material-remixer-js and material-remixer-remote-web projects as well**. It is out of scope for this document to explain how to do it in those projects.
+
+## Adding a new DataType
+
+In order to add a new data type to Remixer, you need to construct a static instance of the DataType class...
+
+```java
+public static final DataType<RT, ST> MY_NEW_TYPE = new DataType<>(
+    "myNewType",
+    RT.class,
+    ST.class,
+    new MyNewTypeValueConverter("myNewType"));
+```
+
+... and register it as a RemixerType somewhere that only gets called once (preferably during your `Application.onCreate`)...
+
+```java
+public class MyApplication extends Application {
+
+  @Override
+  public void onCreate() {
+    super.onCreate();
+    Remixer.registerDataType(MY_NEW_TYPE);
+  }
+}
+```
+
+Now let's dissect the first statement, the `new DataType<>(...)`:
+
+1. It has two generic parameters, RT and ST, the RuntimeType and the SerializedType. The former is used for callbacks during runtime, and the latter is used to serialize, store data and (potentially) sync it to Firebase. They can be exactly the same type, and they usually are.
+2. It has a unique identifier string, in the example it's "myNewType". It can be whatever you want but current identifier strings are "boolean", "color", "number" and "string".
```

Comment:
Is there any guidance on using the oxford comma? I noticed the lack of it in several places